### ISSUE TITLE
ajax should support different content types

### ISSUE THREAD

**Issue by weltan** [pkcsecurity/brevity#5]
currently, ajax component only supports `application/json`. We should default to `application/json` but have an optional param that allows `:content-type` to be set.

**Comment by spencerreeves** [comment]
@weltan Currently this is supported if you pass it in as a header.
ex: `:headers {"Content-Type" "application/pdf"}`

If you look at the code, it merges the Content-Type with the headers, with the headers being the last option. Therefore, if you pass in something else, it will overwrite that field.

**Comment by weltan** [comment]
This is true. But wouldn't it be great if I knew it had that functionality without having to read the code of the component? :)
